api/src/test/java/org/apache/iceberg/expressions/TestInclusiveMetricsEvaluator.java

```diff
@@ -431,4 +433,52 @@ public void testStringStartsWith() {
     shouldRead = new InclusiveMetricsEvaluator(SCHEMA, startsWith("required", aboveMax), true).eval(FILE_4);
     Assert.assertFalse("Should not read: range doesn't match", shouldRead);
   }
+
+  @Test
+  public void testIntegerIn() {
+    boolean shouldRead = new InclusiveMetricsEvaluator(SCHEMA, in("id", 5, 6)).eval(FILE);
+    Assert.assertFalse("Should not read: id below lower bound", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, in("id", 28, 29)).eval(FILE);
+    Assert.assertFalse("Should not read: id below lower bound", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, in("id", 30, 31)).eval(FILE);
+    Assert.assertTrue("Should read: id equal to lower bound", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, in("id", 75, 76)).eval(FILE);
+    Assert.assertTrue("Should read: id between lower and upper bounds", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, in("id", 79, 80)).eval(FILE);
+    Assert.assertTrue("Should read: id equal to upper bound", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, in("id", 80, 81)).eval(FILE);
+    Assert.assertFalse("Should not read: id above upper bound", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, in("id", 85, 86)).eval(FILE);
+    Assert.assertFalse("Should not read: id above upper bound", shouldRead);
+  }
+
+  @Test
+  public void testIntegerNotIn() {
+    boolean shouldRead = new InclusiveMetricsEvaluator(SCHEMA, notIn("id", 5, 6)).eval(FILE);
+    Assert.assertTrue("Should read: id below lower bound", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, notIn("id", 28, 29)).eval(FILE);
+    Assert.assertTrue("Should read: id below lower bound", shouldRead);
+
+    shouldRead = new InclusiveMetricsEvaluator(SCHEMA, notIn("id", 30, 31)).eval(FILE);
```

Comment:
Same as above. If the variables are extracted out, the tests can be in terms of min - 1, min + 1 etc.

Reply:
Updated the `Assert` to improve the readability.

Reply:
I agree with @moulimukherjee, it would be nice for the test cases to be in terms of file's min and max.

Reply:
👌  will update all the tests.